app/api/tests/test_client_resources.py

```diff
@@ -190,7 +190,53 @@ def test_resource_create(
             body=success_response,
             match=[
                 responses.matchers.json_params_matcher(
-                    {"variables": crd.model_dump(exclude=["id"], by_alias=True)},
+                    {
+                        "variables": crd.model_dump(
+                            exclude=["id", "sync_labels"], by_alias=True
+                        )
+                        | {"tags": [tag.model_dump() for tag in resource.tags]}
+                    },
+                    strict_match=False,
+                )
+            ],
+        )
+        result = api_client.resource_create(crd, k8s_metadata)
+        assert result == resource
+
+    def test_resource_create_with_disable_sync_labels(
+        self, test_url, api_client, resource_factory, mocked_responses
+    ):
+        resource = resource_factory(tags=[])
+        crd = resource.to_spec(id=None, sync_labels=False)
+        k8s_metadata = K8sMetadata(
+            name="my-resource",
+            namespace="default",
+            uid="ad0298c5-b84f-4617-b4a2-d3cbbe9f6a4c",
+            labels=resource.to_metadata_labels(),
```

Comment:
This `resource` instance has no tags so this label is empty right? So regardless of `sync_labels` value, the mutation is called with an empty list of tags?

Reply:
AH yes, you're right. We shouldn't have set empty labels here. I updated the test in this commit [1933533](https://github.com/Twingate/kubernetes-operator/pull/606/commits/193353348aac106aaaa82914081c9f2a25c7ff8e). The test should fail when we enable `syncLabels`.